engine/classes/Elgg/Application.php

```diff
@@ -408,6 +408,13 @@ public function run() {
 			return true;
 		}
 
+		if (0 === strpos($path, '/refresh-token')) {
+			$this->loadCore();
+			$this->services->session->start();
```

Comment:
It's fine for now, but I feel like this shouldn't be necessary. Like ElggSession get()/set() should make sure of that.